include/dicom/dicom.h

```diff
@@ -2372,16 +2372,50 @@ void dcm_bot_destroy(DcmBOT *bot);
  */
 
 /**
- * Create a File.
+ * A set of IO functions, see dcm_file_create_io().
+ */
+typedef struct _DcmIO {
+    /** Open an IO object */
+    void *(*open)(DcmError **error, void *client);
+    /** Close an IO object */
+    int (*close)(DcmError **error, void *data);
+    /** Read from an IO object, semantics as POSIX read() */
+    int64_t (*read)(DcmError **error, void *data, char *buffer, int64_t length);
+    /** Seek an IO object, semantics as POSIX seek() */
+    int64_t (*seek)(DcmError **error, void *data, int64_t offset, int whence);
+} DcmIO;
+
+/**
+ * Create a File that reads using a set of DcmIO functions.
+ *
+ * :param io: Set of read functions for this DcmFile
+ * :param client: Client data for read functions
+ *
+ * :return: file
+ */
+DCM_EXTERN
+DcmFile *dcm_file_create_io(DcmError **error, DcmIO *io, void *client); 
+
+/**
+ * Open a file on disk as a DcmFile.
+ *
+ * :param file_path: Path to the file on disk
+ *
+ * :return: file
+ */
+DCM_EXTERN
+DcmFile *dcm_file_open(DcmError **error, const char *file_path);
```

Comment:
I see your point that `dcm_file_create` is misleading, because it does not create a file on disk. However, I still like the `_create` suffix because it indicates that a `DcmFile` object gets created, which ultimately needs to be destroyed.

To clarify the nature of `DcmFile`, I am tempted to rename the type to `DcmFileHandle`, because that term probably better reflects its characteristics. This function could then be called `dcm_filehandle_create`.

Reply:
It would also nicely fit `dcm_filehandle_create_io()`, which conceptually represents an alternative constructor function.

Reply:
`DcmFileHandle` would also already indicate that the object is an abstraction of a file, which may have a representation on disk or just exist in memory.

Reply:
I agree, let's rename it. Could we do that in another PR? I'd like to get this merged so I can merge getframe.

Reply:
Rename PR (but this PR needs merging first) https://github.com/ImagingDataCommons/libdicom/pull/41